docs/guide/custom-ui.md

```diff
@@ -1,39 +1,292 @@
+    modules/star-range-field/ui/apos/components/InputStarRating.vue
+  </template>
+</AposCodeBlock>
+
+In our template element, `AposInputWrapper` takes care of decorating our field with a label, error messages, etc. All we have to do is pass on some standard props that are provided to us. Beyond that, our responsibility is to display the current `value` to the user. We also add event handlers to handle user input, as explained below.
+
+In our script element, we have just two jobs: assigning a new value to `this.next` whenever the value changes, and validating the user's input. The `AposInputMixin` does the rest of the work for us.
+
+To update `this.next`, we implement methods that respond to click events, like the `setValue` and `clear` methods in this example. To validate the user's input, we implement a `validate` method, which accepts the current value and checks constraints like the `required` property of the field. If there is a problem, we return an error code such as `required`, `min` or `max`, otherwise we return `false`. The field configuration is available to us as `this.field`.
+
+The `style` element takes care of CSS for this component. Note that SCSS syntax is available. To avoid conflicts, using the `scoped` attribute is recommended.
+
+::: warning
+If it doesn't seem to work at first, make sure you launched `npm run dev` with the `APOS_DEV=1` environment variable setting as explained earlier. This ensures the Apostrophe admin UI is rebuilt on each code change. You can stop using it when you are finished changing admin UI code.
+:::
+
+### Putting the new schema field type to work
+
+Now we can use the new schema field type in any piece or widget much as we would use an `integer` field:
 
 ```javascript
-// in our modules/announcement/index.js file
+fields: {
+  add: {
+    rating: {
+      type: 'starRating',
+      label: 'Star Rating',
+      required: true
+    }
+  }
+}
+```
+
+The resulting value is then available as the `stars` property of the piece or widget, with an integer value between `1` and `5`.
+
+## Adding custom columns to the piece type manager
+
+Another common extension is a custom column for the piece type manager. By default the manager modal displays the title, certain flags, and the last time the piece was updated. We can extend this to display our custom star ratings.
+
+### Implementing the server-side part
+
+Let's look at a simple piece type module called `review` that takes advantage of the `starRating` field type we defined earlier. We'll add a `rating` field to the schema for the piece type, and also add a column to display it in the manager modal:
+
+<AposCodeBlock>
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  fields: {
+    add: {
+      rating: {
+        type: 'starRating',
+        label: 'Star Rating',
+        required: true
+      }
+    }
+  },
+  columns: {
+    add: {
+      rating: {
+        label: 'Rating',
+        component: 'ColumnStarRating'
```

Comment:
There is no description of what these are. This could be a use case for inline comments with a very brief description of the column properties (optionally using 👇 as I have elsewhere).

Reply:
I'm specifically talking about the label and component properties.